qa/lsan.supp

```diff
@@ -0,0 +1,17 @@
+# leak suppressions needed to run vstart.sh with WITH_ASAN=ON
+# to enable, add this to the environment:
+# LSAN_OPTIONS="suppressions=../qa/lsan.supp"
+
+# from perfglue/heap_profiler.cc
+# gperftools allocates a singleton and never free it
+leak:^MallocExtension::Register
+
+# from src/ceph.in
+leak:^PyObject_Malloc
+leak:^_PyObject_Realloc
+leak:^PyMem_Calloc
+leak:^PyMem_Malloc
+leak:^PyUnicode_New
+leak:^_PyBytes_FromSize
+leak:^_PyObject_GC_Alloc
+leak:^_PyObject_GC_Resize
```

Comment:
had to add this (using libc and rhel8):
```
leak:__interceptor_malloc
```

Reply:
`__interceptor_malloc` is the asan hook that *catches* the leaks. i think you'll want to suppress the function that calls that, instead. ex:
```
Indirect leak of 22079 byte(s) in 23 object(s) allocated from:                                                                                                
    #0 0x7f82e4eba6af in __interceptor_malloc (/usr/lib64/libasan.so.8+0xba6af)                                                                               
    #1 0x7f82e4b03419 in PyObject_Malloc (/lib64/libpython3.8.so.1.0+0x103419)
```
do you have the LeakSanitizer output for the leak you saw? maybe it's the same rgw_admin.cc leak from https://github.com/ceph/ceph/pull/50383?

Reply:
> do you have the LeakSanitizer output for the leak you saw? maybe it's the same rgw_admin.cc leak from #50383?

Possibly related concern: I don't see much overlap with https://github.com/python/cpython/blob/3.11/Misc/valgrind-python.supp.  Are we sure it's not our stuff in `src/pybind` that needs fixing?

Reply:
> Are we sure it's not our stuff in `src/pybind` that needs fixing?

good question, i wasn't able to tell one way or the other from that leak output. i might try running some of the tests under `src/test/pybind/` to see if they show up there too

in general we should only suppress issues from ceph's dependencies, and open tracker issues to get ceph bugs fixed

Reply:
some interesting results.. i tried running `python -m radossss` against several different versions of python. the command immediately fails with `No module named radossss`, and doesn't import any of ceph's modules

https://gist.github.com/cbodley/07a84ad5acf9bb0c8e956f7a6204e659

each python version after 3.8 shows fewer leaks, until 3.12 shows none. so i don't think any of the python leaks i put into `lsan.supp` are specific to stuff under `src/pybind`